posydon/popsyn/synthetic_population.py

```diff
@@ -110,18 +110,19 @@ def merge_parallel_runs(self, path_to_batches):
         if isinstance(path_to_batches, str):
             path_to_batches = [path_to_batches]
         # check if path_to_batches is the same length as the number of metallicities
-        if len(path_to_batches) != len(self.metallicity):
+        if len(path_to_batches) != len(self.metallicities):
             raise ValueError('The number of metallicity and batch directories do not match!')
 
-        for met, path_to_batch in zip(self.metallicity, path_to_batches):
+        for met, path_to_batch in zip(self.metallicities, path_to_batches):
             met_prefix = self.create_met_prefix(met)
             tmp_files = [os.path.join(path_to_batch, f)     \
                          for f in os.listdir(path_to_batch) \
                             if os.path.isfile(os.path.join(path_to_batch, f))]
             
             BinaryPopulation(**self.get_ini_kw()).combine_saved_files(met_prefix+ 'population.h5', tmp_files)
             print(f'Population at Z={met:.2e} Z_sun successfully merged!')
-            os.rmdir(path_to_batch)
+            if len(os.listdir(path_to_batch)) == 0:
+                os.rmdir(path_to_batch)
```

Comment:
Minor, I'd suggest to get here an output on verbose if there is nothing to remove, e.g.
elif self.verbose:
    print(f'{path_to_batch} is not empty')